great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -603,11 +603,10 @@ def list_datasources(self) -> List[dict]:
         """
         datasources: List[dict] = []
         substitutions: dict = self._determine_substitutions()
+
         datasource_name: str
-        for datasource_name in self._datasource_store.list_keys():
-            datasource_config: DatasourceConfig = (
-                self._datasource_store.retrieve_by_name(datasource_name)
-            )
+        datasource_config: DatasourceConfig
```

Comment:
Changes in abstract_data_context.py are from: https://github.com/great-expectations/great_expectations/pull/5572